sonar-batch-protocol/src/main/java/org/sonar/batch/protocol/ProtobufUtil.java

```diff
@@ -44,6 +39,43 @@ private ProtobufUtil() {
     }
   }
 
+  public static <MESSAGE extends Message> Iterable<MESSAGE> readFileMessages(File file, final Parser<MESSAGE> parser) {
+    try {
+      final InputStream inputStream = new BufferedInputStream(new FileInputStream(file));
```

Comment:
InputStream is never closed. You can close it when last "next()" or last "hasNext()" is called. This should do the trick since we are always supposed to read the full content.